contracts/pcv/angle/AngleUniswapPCVDeposit.sol

```diff
@@ -0,0 +1,212 @@
+        uint256 _maxBasisPointsFromPegLP,
+        IStableMaster _stableMaster,
+        IPoolManager _poolManager,
+        IStakingRewards _stakingRewards
+    ) UniswapPCVDeposit(_core, _pair, _router, _oracle, _backupOracle, _maxBasisPointsFromPegLP) {
+        stableMaster = _stableMaster;
+        poolManager = _poolManager;
+        stakingRewards = _stakingRewards;
+        require(_poolManager.token() == address(fei()), "AngleUniswapPCVDeposit: invalid poolManager");
+        require(_stableMaster.agToken() == token, "AngleUniswapPCVDeposit: invalid stableMaster");
+        require(_stakingRewards.stakingToken() == _pair, "AngleUniswapPCVDeposit: invalid stakingRewards");
+
+        // Approve FEI on StableMaster to be able to mint agTokens
+        SafeERC20.safeApprove(IERC20(address(fei())), address(_stableMaster), type(uint256).max);
+        // Approve LP tokens on StakingRewards to earn ANGLE rewards
+        SafeERC20.safeApprove(IERC20(_pair), address(_stakingRewards), type(uint256).max);
+    }
+
+    /// @notice claim staking rewards
+    function claimRewards() external {
+        stakingRewards.getReward();
+    }
+
+    /// @notice mint agToken from FEI held on this contract
+    /// @dev the call will revert if slippage is too high compared to oracle.
+    function mintAgToken(uint256 amountFei)
+        public
+        onlyPCVController
+    {
+        uint256 minAgTokenOut = Decimal.from(amountFei)
+          .div(readOracle())
+          .mul(Constants.BASIS_POINTS_GRANULARITY - maxBasisPointsFromPegLP)
+          .div(Constants.BASIS_POINTS_GRANULARITY)
+          .asUint256();
+
+        uint256 agTokenBalanceBefore = IERC20(token).balanceOf(address(this));
+        stableMaster.mint(
+            amountFei,
+            address(this),
+            poolManager,
+            0
+        );
+        uint256 agTokenBalanceAfter = IERC20(token).balanceOf(address(this));
+        require(agTokenBalanceAfter - agTokenBalanceBefore > minAgTokenOut, "AngleUniswapPCVDeposit: slippage on mint");
+    }
+
+    /// @notice mint agToken from ALL FEI held on this contract
+    /// See mintAgToken(uint256 amount).
+    function mintAgTokenAll()
+        external
+        onlyPCVController
+    {
+        mintAgToken(IERC20(fei()).balanceOf(address(this)));
+    }
+
+    /// @notice burn agToken for FEI
+    /// @dev the call will revert if slippage is too high compared to oracle
+    function burnAgToken(uint256 amountAgToken)
+        public
+        onlyPCVController
+    {
+        uint256 minFeiOut = readOracle() // FEI per X
+          .mul(amountAgToken)
+          .mul(Constants.BASIS_POINTS_GRANULARITY - maxBasisPointsFromPegLP)
+          .div(Constants.BASIS_POINTS_GRANULARITY)
+          .asUint256();
+
+        uint256 feiBalanceBefore = fei().balanceOf(address(this));
+        stableMaster.burn(
+            amountAgToken,
+            address(this),
+            address(this),
+            poolManager,
+            0
```

Comment:
Why not use the slippage here?

Reply:
same rationale as for not using the minimum amount on minting : it's more defensive to check in our own contract than rely on what's done in the external contract